src/supplemental.js

```diff
@@ -0,0 +1,38 @@
+function calcSupplementalDamage(types, damageArray, vals, {remainHP = 1.0, expectedTurn = 1}={}) {
+    for (let supplemental of Object.values(damageArray)) {
+        if (! types.includes(supplemental.type))
+            continue;
+
+        switch (supplemental.type) {
+            case "hp_based":
+                if (!("threshold" in supplemental)) {
+                    console.error("Missing HP threshold in: " + supplemental);
+                    continue;
+                }
+                if (remainHP < supplemental.threshold)
+                    continue;
+                /* FALLTHROUGH */
+            case "other":
+                vals[0] += supplemental.damage;
+                vals[1] += supplemental.damageWithoutCritical;
+                vals[2] += supplemental.ougiDamage;
+                vals[3] += supplemental.chainBurst;
+                continue;
```

Comment:
in this case, "break" is better for switch/case. because if statement has continue.
and logging each iteration for debug. (e.g. add code to display vals changes each iteration, w/o step debugger)